modules/queue/queue.go

```diff
@@ -1,27 +1,62 @@
 // Copyright 2023 The Gitea Authors. All rights reserved.
 // SPDX-License-Identifier: MIT
 
-// Package queue implements a specialized queue system for Gitea.
+// Package queue implements a specialized concurrent queue system for Gitea.
 //
-// There are two major kinds of concepts:
+// Terminology:
 //
-// * The "base queue": channel, level, redis:
-//   - They have the same abstraction, the same interface, and they are tested by the same testing code.
-//   - The dummy(immediate) queue is special, it's not a real queue, it's only used as a no-op queue or a testing queue.
+//  1. Task:
+//     -	A task can be a simple value, such as an integer, or a more complex structure that has multiple fields and
+//     methods. The aim of a task is to be a unit of work, a set of tasks will be sent to a handler to be processed.
```

Comment:
IIRC there is no "task" concept in the queue system. Queue system only handles "item", which is a binary slice.